pymeasure/instruments/common_base.py

```diff
@@ -485,3 +539,23 @@ def setting(set_command, docs,
                                   check_set_errors=check_set_errors,
                                   dynamic=dynamic,
                                   **kwargs)
+
+
+class ChildDescriptor:
```

Comment:
In the beginning, it works as a descriptor, until the Instrument init replaces it with the channels themselves. If I remove `__get__`, it does not work anymore.

Reply:
Ok, just for my curiosity what problem do you have removing `__get__`?

Reply:
> Well, this is not a python descriptor

Yeah it is? It has `__get__`, is used as a class attribute of a managed class, and that attribute gets replaced upon instantiation. What are you missing?

Reply:
Sorry I just don't get the purpose of  this way of using descriptor. Normally `__get__` has two branches one when the descriptor is accessed at class level and another when descriptor is accessed at instance level. Here, the branch at instance level raise an error, but it is never called, because the descriptor is overwritten in the `__init__` and so it's lost. The access at class level, I am not sure it's used.

Reply:
I found the problem in `add_child`, now it works without `__get__`.

Reply:
Should we rename it?
It describes a child (or several).

As it is the first time, I work with descriptors (but not any more), I do not know how fix that name is.

Reply:
I guess this could be leveraged more, I guess Fluent Python Chapter 20 would help in suggesting a better pattern, but I don't have time to dig through it again for some time. :-/

Reply:
Descriptors are normally used to "manage" attributes for an instance (e.g. to do some background verification). I guess they could be used for some housekeeping of the channels here, which maybe could make some code unnecessary, but I'm not sure and it's too late to think well.

Reply:
> I guess this could be leveraged more, I guess Fluent Python Chapter 20 would help in suggesting a better pattern, but I don't have time to dig through it again for some time. :-/

I just realised that we probably _cannot_ lean more on the descriptor interface here -- descriptors are inherently class-based features (like properties), but we want to be able to _dynamically_ change an instance's channel setup (e.g. like for Caspers VNA device). That's a mismatch that is going to show, probably.

Reply:
This case was considered in https://github.com/pymeasure/pymeasure/pull/718#discussion_r1014890666

Reply:
Ah, indeed! Thank you for correcting me!

That will sidestep the channel descriptor, though, so the thing that makes a descriptor "special", i.e. management of some instance attribute via the mechanism you explained

> two branches one when the descriptor is accessed at class level and another when descriptor is accessed at instance level.

will not be possible (as no class-level entity exists for a dynamically created channel).
Therefore, as you said above, there's no purpose in using a descriptor here, anymore.